makeStage2.sh

```diff
@@ -40,5 +41,100 @@ cd $usepop/pop/x/src/
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
 # $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p "${BUILD_HOME}/environments"
+
+# We need to capture the Poplog environment for each build variant. This is
+# required for the poplog command tool, which needs to bind the appropriate 
+# environment variables. Each build variant has slightly different variables
+# and values. So we capture these and later on will synthesise them into 
+# C-code functions. This is the best point at which to capture the variables.
+
+# Note that we do not wish to capture the values of variables automatically
+# introduced by running a shell SHLVL and PWD. Nor do we want to capture
+# the folder location variables poplib, poplocalauto, poplocalbin. That is
+# because GetPoplog has different defaults for those and the old values are
+# irrelevant.
+
+# We run the popenv.sh script inside a clean environment to capture the 
+# set of environment variables needed. We then need to replace any matches
+# of the string "_build/poplog_base" ($usepop) with our unique value USEPOP.
+# This will allow us to dynamically substitute with the selfHome'd value
+# at run-time. 
+#
+# This is inherently a weak strategy because it relies on being able to 
+# identify substitutions of $usepop. We improve its robustness by doing the
+# process twice with different values of $usepop - using the '..' trick.
```

Comment:
This comment about the `..` trick belongs down around line 88 where you first generate two versions of the environment variables, it is at this point the reader needs to understand the context of what it is you're doing.

Reply:
I have changed this a bit & think I have done what is needed.